bin/util.py

```diff
@@ -118,6 +118,7 @@ def read_markdown(parser, path):
     cmd = 'ruby {0}'.format(parser)
     p = Popen(cmd, shell=True, stdin=PIPE, stdout=PIPE,
               close_fds=True, universal_newlines=True)
+    body = str(body.encode('utf-8'))
```

Comment:
Can you explain why this change is necessary? Under what circumstances `body` has different encoding?

Reply:
I explained why this is happening in #445 and why this is necessary to fix the code to run on (any) windows (10, maybe others) machines. I agree it's not elegant, but it's the quickest patch I could think of that should not break in other OSes.

Reply:
there is no explanation in #445.

Reply:
Copied from there:

Then I get the encoding problem. From this [StackOverflow answer](https://stackoverflow.com/a/33291200), this seems to be a by-product of using `universal_newlines=True`. This has also been reported as a bug in Python as well (report [here](https://bugs.python.org/issue27179)). We can get around it by specifying the encoding directly, which is what I did previously, by changing line 121.

Reply:
Can you try replacing `Popen` + `p.communicate` with `subprocess.run(cmd, capture_output=True)`?

Reply:
This works: `p = subprocess.run(cmd, capture_output=True, input=body, text=True, encoding='utf-8')`

Which is just a wrapper for what we had before :)

Reply:
Without the `encoding` parameter, it fails.

Reply:
I see. Let's add `encoding='utf-8'` to the `Popen` call instead of `body = str(body.encode('utf-8'))`.
(btw, why do you need `str(...)`?)

Reply:
Done. 

(`str.encode` returns a byte-string. `str()` converts to a regular text string)

Reply:
> (`str.encode` returns a byte-string. `str()` converts to a regular text string)

oh, right. I usually `.decode()` these.